sdk_contrib/aws_v1_support/src/main/java/io/opentelemetry/sdk/contrib/trace/aws/propagation/AWSXRayPropagator.java

```diff
@@ -0,0 +1,238 @@
+            spanId = parseSpanId(value);
+            break;
+          case SAMPLED_FLAG_KEY:
+            traceFlags = parseTraceFlag(value);
+            break;
+          default:
+            // Ignoring the arbitrary TraceHeader keys in OpenTelemetry
+            break;
+        }
+      }
+
+      if (!traceId.isValid()) {
+        logger.info(
+            "Invalid TraceId in X-Ray trace header: '"
+                + TRACE_HEADER_KEY
+                + "' with value "
+                + traceHeader
+                + "'. Returning INVALID span context.");
+        return SpanContext.getInvalid();
+      }
+
+      if (!spanId.isValid()) {
+        logger.info(
+            "Invalid ParentId in X-Ray trace header: '"
+                + TRACE_HEADER_KEY
+                + "' with value "
+                + traceHeader
+                + "'. Returning INVALID span context.");
+        return SpanContext.getInvalid();
+      }
+
+      return SpanContext
+          .createFromRemoteParent(traceId, spanId, traceFlags, TraceState.getDefault());
+    } catch (Exception e) {
+      logger.log(Level.INFO, "Error parsing X-Ray trace header. "
+          + "Returning INVALID span context.", e);
+      return SpanContext.getInvalid();
+    }
+  }
+
+  @Nullable
+  private static String[] extractKeyAndValue(String keyValueString) {
+    keyValueString = keyValueString.trim();
+    if (StringUtils.isNullOrEmpty(keyValueString)) {
+      throw new IllegalArgumentException("Invalid key value pair: " + keyValueString);
+    }
+
+    String[] keyAndValue = keyValueString.split(String.valueOf(KV_DELIMITER));
+    if (keyAndValue.length != 2) {
+      throw new IllegalArgumentException("Invalid key value pair: " + keyValueString);
+    }
+
+    if (keyAndValue[0].isEmpty() || keyAndValue[1].isEmpty()) {
+      throw new IllegalArgumentException("Invalid key value pair: " + keyValueString);
+    }
+
+    return keyAndValue;
+  }
+
+  @SuppressWarnings("StringSplitter")
+  private static TraceId parseTraceId(String xrayTraceId) {
+    String[] parts = xrayTraceId.split(String.valueOf(TRACE_ID_DELIMITER));
+    if (parts.length != 3) {
+      return TraceId.getInvalid();
+    }
+
+    if (!parts[0].equals(TRACE_ID_VERSION) ||
+        parts[1].length() != TRACE_ID_FIRST_PART_LENGTH ||
+        parts[2].length() != TRACE_ID_SECOND_PART_LENGTH) {
+      return TraceId.getInvalid();
+    }
+
+    // X-Ray trace id format is 1-{8 digit hex}-{24 digit hex}
+    return TraceId.fromLowerBase16(parts[1] + parts[2], 0);
+  }
+
+  private static SpanId parseSpanId(String xrayParentId) {
+    if (xrayParentId.length() != PARENT_ID_LENGTH) {
+      return SpanId.getInvalid();
+    }
+
+    return SpanId.fromLowerBase16(xrayParentId, 0);
+  }
+
+  private static TraceFlags parseTraceFlag(String xraySampledFlag) {
+    if (xraySampledFlag.length() != SAMPLED_FLAG_LENGTH) {
+      throw new IllegalArgumentException("Invalid Sampling flag: " + xraySampledFlag);
+    }
+
```

Comment:
```java
char flag = xraySampledFlag.charAt(0);
if (flag == IS_SAMPLED)
etc
```

Reply:
Will fix in rev2. But I want to keep the length check since `Sampled=123` is invalid in X-Ray context.